snapshotter.go

```diff
@@ -85,41 +84,38 @@ func (s *ShardSnapshotter) SnapshotShard() (*Snapshot, error) {
 		latest = l
 	}
 
-	if latest != nil {
```

Comment:
There didn't seem to be a way for latest to be nil at this point if above bits didn't error.